samza-core/src/main/scala/org/apache/samza/system/SystemConsumers.scala

```diff
@@ -172,7 +172,10 @@ class SystemConsumers (
   def start {
     for ((systemStreamPartition, offset) <- sspToRegisteredOffsets.asScala) {
       val consumer = consumers(systemStreamPartition.getSystem)
-      consumer.register(systemStreamPartition, offset)
+      // If elasticity is enabled then the RunLoop gives SSP with keybucket
+      // but the actual systemConsumer which consumes from the input does not know about KeyBucket.
+      // hence, use an SSP without KeyBucket
+      consumer.register(new SystemStreamPartition(systemStreamPartition.getSystemStream, systemStreamPartition.getPartition), offset)
```

Comment:
Is there another way we can do this, adding this discrepancy adds additional confusion

Reply:
working on this. trying to find a way that will work well for the next planned pr of existing container metrics reporting .
will update soon.